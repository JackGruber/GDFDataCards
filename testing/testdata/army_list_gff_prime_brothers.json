{
    "gameSystem": "gff",
    "armyId": "oqnnu0gk8q6hyyny",
    "armyIds": [
        "oqnnu0gk8q6hyyny"
    ],
    "armyFaction": "Prime Brothers",
    "armyName": "Prime Brothers",
    "modified": "2023-10-14T14:12:01.653Z",
    "saveVersion": 3,
    "listPoints": 350,
    "list": {
        "id": "UH-dFYSwXVZi",
        "key": "3gLSbzEApWjwko4cWvWzy",
        "name": "Prime Brothers",
        "units": [
            {
                "id": "Z2WiRfh",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "8hK_N",
                "selectedUpgrades": [
                    {
                        "instanceId": "AcNhrecuJ",
                        "upgradeId": "BCvkhSe",
                        "optionId": "fwO6jFd"
                    },
                    {
                        "instanceId": "WD2CyBmQF",
                        "upgradeId": "MovgoCq",
                        "optionId": "d6wOTSF"
                    },
                    {
                        "instanceId": "W9QLUGgl_",
                        "upgradeId": "80QB0Hq",
                        "optionId": "cd-2wsN"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "hvNiOwU",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "Q7qdQ",
                "selectedUpgrades": [
                    {
                        "instanceId": "A_NOsLoyO",
                        "upgradeId": "qDSztwg",
                        "optionId": "m9jFfBF"
                    },
                    {
                        "instanceId": "Hm7SLybjg",
                        "upgradeId": "8l4r8",
                        "optionId": "kaY44"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "NeuN0Gx",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "kmMjq",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "0PbBL8m",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "z_4pO",
                "selectedUpgrades": [
                    {
                        "instanceId": "bFPv-yqRX",
                        "upgradeId": "rH5Mlxw",
                        "optionId": "Fj2ZT9R"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            }
        ],
        "points": 350,
        "pointsLimit": 300,
        "creationTime": "1697292673994",
        "campaignMode": false,
        "competitive": true,
        "gameSystem": "gff",
        "isCloud": false,
        "modified": "2023-10-14T14:12:01.653Z",
        "selectedUnitId": "z_4pO",
        "unitPreview": null
    },
    "favourite": false,
    "armyVersions": [
        {
            "version": "3.1.0",
            "armyId": "oqnnu0gk8q6hyyny"
        }
    ]
}