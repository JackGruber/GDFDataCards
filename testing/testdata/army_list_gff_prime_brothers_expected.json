{
    "armyId": "oqnnu0gk8q6hyyny",
    "gameSystem": "gff",
    "gameSystemId": 3,
    "armyName": "Prime Brothers",
    "listPoints": 350,
    "listName": "Prime Brothers",
    "units": [
        {
            "type": "Prime Master",
            "name": "Prime Master",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "Z2WiRfh",
            "cost": 75,
            "defense": 2,
            "quality": 3,
            "upgrades": [
                "A1",
                "B1",
                "C1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 2,
                    "count": 1,
                    "name": "Master Heavy Pistol",
                    "range": 12,
                    "specialRules": [],
                    "ap": "1"
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Relic Sword",
                    "specialRules": [
                        {
                            "name": "Deadly",
                            "rating": 3,
                            "label": "Deadly(3)"
                        }
                    ],
                    "ap": 2
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "hero",
                    "name": "Hero",
                    "rating": "",
                    "label": "Hero"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "3",
                    "label": "Tough(3)"
                }
            ],
            "upgradeCost": [
                50,
                15,
                10
            ],
            "equipment": [
                {
                    "name": "Captain",
                    "specialRules": [
                        {
                            "key": "battle-rites",
                            "name": "Battle Rites",
                            "type": "ArmyBookRule",
                            "label": "Battle Rites",
                            "rating": ""
                        }
                    ]
                },
                {
                    "name": "Combat Shield",
                    "specialRules": [
                        {
                            "key": "heavy shield",
                            "name": "Heavy Shield",
                            "type": "ArmyBookRule",
                            "label": "Heavy Shield",
                            "modify": false
                        }
                    ]
                }
            ]
        },
        {
            "type": "Assault Prime Brother",
            "name": "Assault Prime Brother",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "hvNiOwU",
            "cost": 30,
            "defense": 2,
            "quality": 3,
            "upgrades": [
                "D1",
                "G1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Heavy Pistol",
                    "range": 12,
                    "specialRules": [],
                    "ap": "1"
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Energy Hammer",
                    "specialRules": [
                        {
                            "name": "Blast",
                            "rating": 3,
                            "label": "Blast(3)"
                        }
                    ]
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "veteran-infantry",
                    "name": "Veteran Infantry",
                    "type": "ArmyBookRule",
                    "label": "Veteran Infantry",
                    "rating": "",
                    "count": 1
                }
            ],
            "upgradeCost": [
                10,
                5
            ]
        },
        {
            "type": "Infernal",
            "name": "Infernal",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "NeuN0Gx",
            "cost": 40,
            "defense": 2,
            "quality": 3,
            "upgrades": [],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Flamer",
                    "range": 12,
                    "specialRules": [
                        {
                            "key": "blast",
                            "name": "Blast",
                            "rating": "3",
                            "modify": false,
                            "label": "Blast(3)"
                        },
                        {
                            "key": "reliable",
                            "name": "Reliable",
                            "rating": "",
                            "modify": false,
                            "label": "Reliable"
                        }
                    ]
                }
            ],
            "specialRules": [
                {
                    "id": 65,
                    "key": "fearless",
                    "name": "Fearless",
                    "label": "Fearless"
                },
                {
                    "id": 74,
                    "key": "relentless",
                    "name": "Relentless",
                    "label": "Relentless"
                }
            ]
        },
        {
            "type": "Eliminator",
            "name": "Eliminator",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "0PbBL8m",
            "cost": 60,
            "defense": 3,
            "quality": 3,
            "upgrades": [
                "K1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Laser Sniper Rifle",
                    "range": 30,
                    "specialRules": [
                        {
                            "key": "deadly",
                            "name": "Deadly",
                            "type": "ArmyBookRule",
                            "label": "Deadly(3)",
                            "modify": false,
                            "rating": 3
                        },
                        {
                            "key": "sniper",
                            "name": "Reliable",
                            "type": "ArmyBookRule",
                            "label": "Reliable",
                            "modify": false
                        }
                    ],
                    "ap": 2
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "scout",
                    "name": "Scout",
                    "rating": "",
                    "label": "Scout"
                },
                {
                    "key": "stealth",
                    "name": "Stealth",
                    "rating": "",
                    "label": "Stealth"
                }
            ],
            "upgradeCost": [
                55
            ]
        }
    ]
}